core/src/main/java/org/springframework/security/event/AuthenticationFailureEventKeyValuesProvider.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.event;
+
+import io.micrometer.common.KeyValues;
+import io.micrometer.observation.Observation;
+
+import org.springframework.security.authentication.event.AbstractAuthenticationFailureEvent;
+
+/**
+ * A {@link io.micrometer.observation.Observation.KeyValuesProvider} for
+ * {@link org.springframework.security.access.event.AuthorizationFailureEvent}s
+ *
+ * @author Josh Cummings
+ * @since 6.0
+ */
+public final class AuthenticationFailureEventKeyValuesProvider
+		implements Observation.KeyValuesProvider<SecurityEventObservationContext<AbstractAuthenticationFailureEvent>> {
```

Comment:
Sounds great. I updated the authentication observation to use `ObservationConvention`. As for the event handling, I think since I'm now calling `observation.event`, these classes don't make as much sense anymore as-is.

Please see `ObservationSecurityEventListener` for details. Essentially, an observation security event can hold a `KeyValues` instance to indicate which are the relevant attributes of the underlying `SecurityEvent` to publish.